src/main/java/uk/gov/laa/gpfd/controller/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.laa.gpfd.controller;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+import uk.gov.laa.gpfd.exception.CsvStreamException;
+import uk.gov.laa.gpfd.exception.DatabaseReadException;
+import uk.gov.laa.gpfd.exception.ReportIdNotFoundException;
+import uk.gov.laa.gpfd.model.ReportIdGet400Response;
+import uk.gov.laa.gpfd.model.ReportsGet500Response;
+
+import static org.springframework.http.ResponseEntity.badRequest;
+import static org.springframework.http.ResponseEntity.internalServerError;
+
+@Slf4j
+@ControllerAdvice
+public class GlobalExceptionHandler {
```

Comment:
Can these be tested?

Reply:
Thank you for the great suggestion! You’re absolutely right; I’ve added the necessary test cases to cover this now. It was a small refactor, and I overlooked adding the test initially—my apologies for that. Please let me know if there’s anything else you’d like me to address!